src/Microsoft.AspNet.OData.Shared/Query/ExpandQueryBuilder.cs

```diff
@@ -40,21 +40,45 @@ public virtual string GenerateExpandQueryParameter(object value, IEdmModel model
 
         private string GenerateExpandQueryStringInternal(object value, IEdmModel model, bool isNestedExpand)
         {
+            string expandString = "";
+
+            if (value == null || model == null)
+            {
+                return expandString;
+            }
+
             Type type = value.GetType();
             bool isCollection = TypeHelper.IsCollection(type, out Type elementType);
             IEnumerable collection = null;
+
             if (isCollection)
             {
                 type = elementType;
                 collection = value as IEnumerable;
             }
 
             string edmFullName = type.EdmFullName();
+
+            if (string.IsNullOrEmpty(edmFullName))
```

Comment:
This check is not superfluous. Given how the `EdmFullName` helper method is implemented, it cannot return null or empty.